kornia/utils/dataset.py

```diff
@@ -0,0 +1,74 @@
+from typing import Optional
+
+from torch import Tensor
+from torch.utils.data import Dataset
+
+
+class CachedDataSet(Dataset):
+    """
+    Args:
+        caching: caching methods like in-memroy, mldb, redis, spark, etc.
+        resize_to: if to save the resized images.
+        backend: torch, cv2, rust etc.
+        device: save the image directly to cpu or gpu.
+        dtype: blabalba
+    """
+
+    def __init__(
+        self,
+        caching: str = "in-memory",
+        resize_to: Optional[tuple] = None,
+        backend: str = "torch",
+        device: str = None,
+        dtype=None,
+        transform=None
+    ) -> None:
+        super().__init__()
+        self._device = device
+        self._dtype = dtype
+        self.resize_to = resize_to
+        self.backend = backend
+        self.transform = transform
+
+        if caching == "in-memory":
+            self.cache = InMemoryCache()
+        else:
+            raise ValueError
+
+    def __len__(self,):
+        return len(self.data)
+
```

Comment:
As a pointer, this will implement 
https://github.com/kornia/kornia-rs/blob/main/test/test_io.py#L32

Mostly ready and tested internally. Working on the packaging now